### PR TITLE
MM-42677 Adding resource to create TGW & share it to external aws acc…

### PR DESCRIPTION
**Pull request by Mshahidtaj**
#### Summary
Adding a module to create a Transit gateway & share it as RAM resource for cross-account sharing.
#### Ticket Link
(https://mattermost.atlassian.net/browse/MM-42677)

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by stafot — DISMISSED** [review]

File: aws/transit-gateway-setup/variables.tf [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  description = "Whether resource attachment requests are automatically accepted"
+  type        = string
+  default     = "enable"
+}
+
+variable "enable_default_route_table_association" {
+  description = "Whether resource attachments are automatically associated with the default association route table"
+  type        = string
+  default     = "enable"
+}
```
Comment (angeloskyratzakos) [inline aws/transit-gateway-setup/variables.tf:43]: maybe these can be boolean type?

File: aws/transit-gateway-setup/variables.tf [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  description = "Description of the EC2 Transit Gateway"
+  type        = string
+  default     = null
+}
+
+variable "enable_dns_support" {
+  description = "Should be true to enable DNS support in the TGW"
+  type        = string
+  default     = "enable"
+}
```
Comment (angeloskyratzakos) [inline aws/transit-gateway-setup/variables.tf:55]: same here

File: aws/transit-gateway-setup/variables.tf [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  description = "Whether VPN Equal Cost Multipath Protocol support is enabled"
+  type        = bool
+  default     = false
+}
+
+variable "enable_default_route_table_propagation" {
+  description = "Whether resource attachments automatically propagate routes to the default propagation route table"
+  type        = string
+  default     = "enable"
+}
```
Comment (angeloskyratzakos) [inline aws/transit-gateway-setup/variables.tf:67]: and here

File: aws/transit-gateway-setup/variables.tf (hunk 1 above)
Comment (Mshahidtaj) [reply aws/transit-gateway-setup/variables.tf:43]: Hi @angeloskyratzakos these are string types as part of TF `aws_ec2_transit_gateway`, Reference https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_transit_gateway#auto_accept_shared_attachments

File: aws/transit-gateway-setup/variables.tf (hunk 2 above)
Comment (Mshahidtaj) [reply aws/transit-gateway-setup/variables.tf:55]: Hi @angeloskyratzakos these are string types as part of TF `aws_ec2_transit_gateway`, Reference https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_transit_gateway#auto_accept_shared_attachments

File: aws/transit-gateway-setup/variables.tf (hunk 3 above)
Comment (Mshahidtaj) [reply aws/transit-gateway-setup/variables.tf:67]: Hi @angeloskyratzakos these are string types as part of TF `aws_ec2_transit_gateway`, Reference https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_transit_gateway#auto_accept_shared_attachments

**Review by angeloskyratzakos — DISMISSED** [review]

**Review by spirosoik — CHANGES_REQUESTED** [review]
we need to generate the `README.md` (you can use the `terraform-docs`) also I would like us to think as it's open source if this module is reusable and not only MM coupled.

Great job for the `tflint` thanks a lot

**Comment by Mshahidtaj** [comment]
> we need to generate the `README.md` (you can use the `terraform-docs`) also I would like us to think as it's open source if this module is reusable and not only MM coupled.
> 
> Great job for the `tflint` thanks a lot

Yeah, 100%  Agreed & added a Readme file for it.

**Review by angeloskyratzakos — APPROVED** [review]

**Review by spirosoik — APPROVED** [review]

**Pull request merged**